odoo/models.py

```diff
@@ -190,6 +194,24 @@ def check_company_domain_parent_of(self, companies):
     ])]
 
 
+def check_companies_domain_parent_of(self, companies):
+    """ A `_check_company_domain` function that lets a record be used if
+        any company in record.company_ids is a parent of any of the given companies.
+    """
+    if isinstance(companies, str):
+        return [('company_ids', 'parent_of', companies)]
+
+    companies = [id for id in to_company_ids(companies) if id]
+    if not companies:
+        return []
+
+    return ['|', ('company_ids', '=', False), ('company_ids', 'in', [
+        int(parent)
+        for rec in self.env['res.company'].sudo().browse(companies)
+        for parent in rec.parent_path.split('/')[:-1]
+    ])]
```

Comment:
Having a quick look at the code, I'm not sure the domain parser is actually able to handle that for a `many2many` field, since it would need to generate a query that does both an `EXISTS (SELECT ...)` and a `NOT EXISTS (SELECT ...)`

But in any case, I think the only reason I added the False case is because I misguidedly thought it was necessary for `account.cash.rounding`. I'll just remove it.